### DISCUSSION TITLE
Paragraph terminators for text selection in TEdit

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1662] [Q&A]
Is the period character `.` supposed to be a paragraph terminator as far as text selection in TEdit is concerned?

Consider this Medley Online session in which I typed a block of text continuosly and pressed `ENTER` only once at the end of the paragraph. The text contained two sentences terminated by `.` characters. With the caret at the beginning of the blank line immediately after the paragraph, when I middle-clicked the first line of the pargraph on the liner bar only the last two lines were selected:

![tedit-selection1](https://github.com/Interlisp/medley/assets/10342479/bf4dbc08-25ac-42b8-af12-7dfe691a548c)

Then I typed another paragraph terminated by pressing `ENTER` once at the end, this time with one sentence. With the caret at the beginning of the blank line immediately after the paragraph, clicking the first line on the line bar selected the full paragraph:

![tedit-selection2](https://github.com/Interlisp/medley/assets/10342479/93fa54ce-df74-4ebc-8af8-0aee56fa65d8)

I would expect the full paragraph to be selected in both cases.

**Comment by rmkaplan** [comment]
I see this too, even in my more advanced version.  There still appear to be some alignment/synchronization issues.

**Comment by pamoroso** [comment]
Triple mid-clicking a word in the text region selects only the word and not the line, which is likely related.

**Comment by rmkaplan** [comment]
TEDIT doesn’t (yet) support double or triple clicking. That was a later innovation, better for one button mice. Tedit uses the right-button-extend paradigm. Sent from my iPhoneOn Apr 15, 2024, at 9:38 AM, Paolo Amoroso ***@***.***> wrote:﻿
Triple mid-clicking a word in the text region selects only the word and not the line, which is likely related.
